components/admin_real/desktop/ReservationFieldsModal.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from "react";
+import Modal from "antd/lib/modal/Modal";
+import { Alert, DatePicker, Input } from "antd";
+import { TimePicker } from "antd";
+import moment from "moment";
+import { Select } from "antd";
+
+const { Option } = Select;
+
+const ReservationFieldsModal = ({
+  reservation = {},
+  onSubmit,
+  modalTitle,
+  modalOpen = false,
+  setModalOpen,
+}) => {
+  const dateFormat = "YYYY-MM-DD";
+  const timeFormat = "HH:mm";
+  const id = reservation.id || 123;
+
+  const [date, setDate] = useState(reservation.date);
+  const [time, setTime] = useState(reservation.time || 0);
+  const [certified, setCertified] = useState(
+    reservation.diver_certified ? reservation.diver_certified.toString() : null
+  );
+  const [reservationType, setReservationType] = useState(
+    reservation.reservation_type
+  );
+  const [name, setName] = useState(
+    reservation.diver_information ? reservation.diver_information.name : null
+  );
+  const [lastName, setLastName] = useState(
+    reservation.diver_information
+      ? reservation.diver_information.lastName
+      : null
+  );
+  const [age, setAge] = useState(
+    reservation.diver_information ? reservation.diver_information.age : null
+  );
+  const [experience, setExperience] = useState(reservation.number_of_dives);
+  const [email, setEmail] = useState(
+    reservation.diver_information ? reservation.diver_information.email : null
+  );
+
+  const reservationValid = () => {
+    if (
+      date &&
+      time &&
+      certified &&
+      reservationType &&
+      experience &&
+      name &&
+      lastName &&
+      age &&
+      email
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  return (
+    <div>
+      <Modal
+        title={modalTitle}
+        visible={modalOpen}
+        onOk={() => {
+          if (reservationValid()) {
+            const { data, error } = onSubmit({
```

Comment:
are we sure we can do this? we are probably not getting the values back right? this is an async function, so we need to do the `.then`, instead of the `const {data, error} = `. 

i am pretty sure the values are undefined now because this is an async function

Reply:
yep that's my bad, did this part kind of quick